conans/model/build_info.py

```diff
@@ -111,10 +191,75 @@ def __init__(self, root_folder):
         self.libdirs.append(DEFAULT_LIB)
         self.bindirs.append(DEFAULT_BIN)
         self.resdirs.append(DEFAULT_RES)
-        self.builddirs.append("")
+        self.builddirs.append(DEFAULT_BUILD)
         # public_deps is needed to accumulate list of deps for cmake targets
         self.public_deps = []
         self.configs = {}
+        self._deps = OrderedDict()
+
+    def _check_dirs_values(self):
+        default_dirs_mapping = {
+            "includedirs": [DEFAULT_INCLUDE],
+            "libdirs": [DEFAULT_LIB],
+            "bindirs": [DEFAULT_BIN],
+            "resdirs": [DEFAULT_RES],
+            "builddirs": [DEFAULT_BUILD],
+            "srcdirs": []
+        }
+        msg_template = "Using Components and global '{}' values ('{}') is not supported"
+        for dir_name in ["includedirs", "libdirs", "bindirs", "builddirs"]:
+            dirs_value = getattr(self, dir_name)
+            if dirs_value is not None and dirs_value != default_dirs_mapping[dir_name]:
+                raise ConanException(msg_template.format(dir_name, dirs_value))
+
+    def _clear_dirs_values(self):
+        default_dirs_mapping = {
+            "includedirs": [DEFAULT_INCLUDE],
+            "libdirs": [DEFAULT_LIB],
+            "bindirs": [DEFAULT_BIN],
+            "resdirs": [DEFAULT_RES],
+            "builddirs": [DEFAULT_BUILD],
+            "srcdirs": []
+        }
+        for dir_name in ["includedirs", "libdirs", "bindirs", "builddirs"]:
+            if getattr(self, dir_name) == default_dirs_mapping[dir_name]:
+                self.__dict__[dir_name] = None
+
+    @property
+    def libs(self):
+        if self._deps:
+            deps = [v for v in self._deps.values()]
+            deps_sorted = sorted(deps, key=lambda component: len(component.deps))
+            result = []
+            for dep in deps_sorted:
+                for sys_dep in dep.system_deps:
+                    if sys_dep and sys_dep not in result:
+                        result.append(sys_dep)
+                if dep.lib:
+                    result.append(dep.lib)
+            return result
+        else:
+            return self._libs
```

Comment:
There is already a `libs` property in `_CppInfo`, which is the parent class. There you are not adding `system_deps` and here you are. I will come back to this once I reach the tests...